test/nodejs/streamaggr2.js

```diff
@@ -1289,20 +1289,204 @@ describe('Simple linear regression test', function () {
+                lambda: 50,
+                alpha: 0.9999
+            };
+            store = base.store("Store");
+            pht = store.addStreamAggr(aggrPHT);
+        });
+
+        after(function() {
+            base.close();
+        });
+
+        it("should initialize stream aggregate and return default values", function() {
+            assert.deepEqual(pht.saveJson(), {
+                minInstances: 30,
+                delta: 0.005,
+                alpha: 0.9999,
+                lambda: 50,
+                drift: 0,
+                driftOffset: -1
+            });
+        });
+
+        it ('should not detect concept drift', function() {
+            // creating start time
+            let time = new Date();
+            let timeStr = time.toISOString();
+            let changes = 0;
+            time.setSeconds(time.getSeconds() - 1000);
+
+            for (let i = 0; i < 1000; i++) {
+                // add one second to the timestamp and create an ISO string
+                time.setSeconds(time.getSeconds() + 1);
+                let timeStr = time.toISOString();
+                // create value
+                let value = Math.random() * 1;
+                // adding values to the signal store
+                store.push({ Time: timeStr, Value: value });
+                // counting changes
+                if (pht.saveJson().drift == 1) {
+                    changes++;
+                }
+            }
+            // checking if drift has been correctly detected
+            assert.equal(changes, 0);
+            driftOffset = pht.saveJson().driftOffset;
+            assert.equal(driftOffset, -1);
+        });
+
+        it('should detect concept drift in proximity of simulated change', function() {
+            // creating start time
+            let time = new Date();
+            let timeStr = time.toISOString();
+            let changes = 0;
+
+            // simulating concept drift at element 1000 in a time series
+            for (let i = 0; i < 2000; i++) {
+                // add one second to the timestamp and create an ISO string
+                time.setSeconds(time.getSeconds() + 1);
+                let timeStr = time.toISOString();
+                // create value
+                let value = Math.random() * 1;
+                if (i > 1000) {
+                    value = Math.random() * 2 + 1;
+                }
+                // adding values to the signal store
+                store.push({ Time: timeStr, Value: value });
+                // counting changes
+                if (pht.saveJson().drift == 1) {
+                    changes++;
+                }
+            }
+            // checking if drift has been correctly detected
+            assert.equal(changes, 1);
+            driftOffset = pht.saveJson().driftOffset;
+            assert.equal((driftOffset < 1000) && (driftOffset > 900), true);
+        })
+
+        it('should save and then load the aggregate', function() {
```

Comment:
add two tests:
- save/load uninitialized aggregate
- save/load initialized aggregate with data

Reply:
Example: https://github.com/rupnikj/qminer/blob/master/test/nodejs/streamaggr.js#L6080

Reply:
Made 2 tests. Is this OK?